logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+			for( int i = 0; i < BLOB_MAX_FILES; i++){
+				char *filename = mallocz(snprintf(NULL, 0, BLOB_STORE_FILENAME ".%d", i) + 1);
+				sprintf(filename, BLOB_STORE_FILENAME ".%d", i);
+				rc = sqlite3_bind_text(stmt_init_BLOBS_table, 1, filename, -1, NULL);
+                if (rc != SQLITE_OK) fatal_sqlite3_err(rc, __LINE__);
+                rc = sqlite3_bind_int64(stmt_init_BLOBS_table, 2, (sqlite3_int64) 0);
+                if (rc != SQLITE_OK) fatal_sqlite3_err(rc, __LINE__);      
+				rc = sqlite3_step(stmt_init_BLOBS_table);
+				if (rc != SQLITE_DONE) fatal_sqlite3_err(rc, __LINE__);
+				rc = sqlite3_reset(stmt_init_BLOBS_table);
+				if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+				freez(filename);
+			}
+			rc = sqlite3_finalize(stmt_init_BLOBS_table);
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+		}
+		rc = sqlite3_finalize(stmt_check_if_BLOBS_TABLE_exists);
+		if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+		
+		/* If LOGS_TABLE doesn't exist, create it */
+		rc = sqlite3_exec(p_file_infos_arr->data[i]->db,
+                      "CREATE TABLE IF NOT EXISTS " LOGS_TABLE "("
+                      "Id 					INTEGER 	PRIMARY KEY,"
+                      "FK_BLOB_Id			INTEGER		NOT NULL,"
+                      "BLOB_Offset			INTEGER		NOT NULL,"
+                      "Timestamp 			INTEGER		NOT NULL,"
+                      "Msg_compr_size 		INTEGER		NOT NULL,"
+                      "Msg_decompr_size 	INTEGER		NOT NULL,"
+                      "FOREIGN KEY (FK_BLOB_Id) REFERENCES " BLOBS_TABLE " (Id) ON DELETE CASCADE ON UPDATE CASCADE"
+                      ");",
+                      0, 0, &err_msg);
+		if (unlikely(rc != SQLITE_OK)) {
+			sqlite3_free(err_msg);
+			fatal("Failed to create %s. SQL error: %s\n", LOGS_TABLE, err_msg);
+		} 
+		// else debug(D_LOGS_MANAG, "Table %s created successfully\n", LOGS_TABLE);
+		
+		// Create index on LOGS_TABLE Timestamp
+		/* TODO: If this doesn't speed up queries, check SQLITE R*tree module. 
+		 * Requires benchmarking with/without index. */
+		rc = sqlite3_exec(p_file_infos_arr->data[i]->db,
+						  "CREATE INDEX IF NOT EXISTS logs_timestamps_idx "
+						  "ON " LOGS_TABLE "(Timestamp);",
+						  0, 0, &err_msg);
+		if (unlikely(rc != SQLITE_OK)) {
+			fatal("Failed to create logs_timestamps_idx. SQL error: %s", err_msg);
+			sqlite3_free(err_msg);
+		} 
+		// else debug(D_LOGS_MANAG, "logs_timestamps_idx created successfully\n");
+
+		/* Remove excess BLOBs beyond BLOB_MAX_FILES (from both DB and disk 
+		 * storage). This is useful if BLOB_MAX_FILES is reduced after an agent
+		 * restart (for example, if in the future it is not hardcoded, but 
+		 * instead it is read from the configuration file). LOGS_TABLE entries
+		 * should be deleted automatically (due to ON DELETE CASCADE). */
+		{
+			sqlite3_stmt *stmt_get_BLOBS_TABLE_size;
```

Comment:
Like previous function, I also suggest to split this function and keep isolated `prepare` and `execute`.